File: DEVBGDEMO/DEVBGDEMO/Steps/ApiSteps.cs
using Reqnroll;
using RestSharp;
using FluentAssertions;
using DEVBGDEMO.Hooks;
using System.Net;
using NUnit.Framework;

namespace DEVBGDEMO.Steps
{
    [Binding]
    public class ApiSteps
    {
        private static string BASE_API_URL = "https://demoqa.com";

        [Given("a user exists via the API")]
        public static void GivenUserExistsViaApi()
        {
            var email = "poliom";
            var password = "QWerty!@34";

            var client = new RestClient(BASE_API_URL);
            var request = new RestRequest("/Account/v1/Login", Method.Post);
            request.AddHeader("Content-Type", "application/json");
            var body = @"{""userName"":""" + email + @""",""password"":""" + password + @"""}";
            request.AddStringBody(body, DataFormat.Json);
            var response = client.Execute(request);
            var content = response.Content;

            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK));

            UiHooks.CreatedEmail = email;
            UiHooks.CreatedPassword = password;
        }
    }
}

File: DEVBGJAVADEMO/DEVBGJAVADEMO/src/test/java/steps/ApiSteps.java
package steps;

import hooks.UiHooks;
import io.cucumber.java.en.Given;
import io.restassured.http.ContentType;
import io.restassured.response.Response;
import static io.restassured.RestAssured.given;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class ApiSteps {
    private static final String BASE_API_URL = "https://demoqa.com";

    @Given("a user exists via the API")
    public void aUserExistsViaTheApi() {
        String email = "poliom";
        String password = "QWerty!@34";

        String json = "{\"userName\":\"" + email + "\",\"password\":\"" + password + "\"}";

        Response response =
                given()
                        .baseUri(BASE_API_URL)
                        .contentType(ContentType.JSON)
                        .body(json)
                        .when()
                        .post("/Account/v1/Login");

        String content = response.asString();
        assertEquals(200, response.statusCode(), content);

        UiHooks.createdEmail = email;
        UiHooks.createdPassword = password;
    }
}
